poseidon-paramgen/src/matrix/traits.rs

```diff
@@ -1,34 +1,13 @@
-use core::slice::Chunks;
-
 use anyhow::Result;
-use ark_std::vec::Vec;
 
 /// Basic matrix operations all matrices should implement.
 pub trait MatrixOperations<F> {
```

Comment:
I would merge it with `BasicMatrixOperations` or rename it to `MatrixOperationsExt` and derive from the former

Reply:
If there are cases that implement `BasicMatrixOperations` without `MatrixOperationsExt` then I'd go with renaming to `MatrixOperationsExt`. In the other case, I'd merge.